From fd3f5b44af00102d80368aba6fe28b3cfce780be Mon Sep 17 00:00:00 2001
From: Stivvo <stivvo01@gmail.com>
Date: Mon, 19 Oct 2020 12:11:43 +0200
Subject: [PATCH] make (un)install and .desktop file

By default installs the executable in /usr/local/bin and the .desktop
file in /usr/share/wayland-sessions.

Example with different paths, executable in /usr/bin, .desktop in
/usr/share/sessions :

sudo make install -e DESKTOP=/usr/share/sessions PREFIX=/usr
---
 Makefile    | 11 +++++++++++
 dwl.desktop |  5 +++++
 2 files changed, 16 insertions(+)
 create mode 100644 dwl.desktop

diff --git a/Makefile b/Makefile
index 56ab7512..6d5c1e1f 100644
--- a/Makefile
+++ b/Makefile
@@ -9,6 +9,9 @@ PKGS = wlroots wayland-server xcb xkbcommon
 CFLAGS += $(foreach p,$(PKGS),$(shell pkg-config --cflags $(p)))
 LDLIBS += $(foreach p,$(PKGS),$(shell pkg-config --libs $(p)))
 
+PREFIX?= /usr/local
+BINDIR?= $(PREFIX)/bin
+DESKTOP?= /usr/share/wayland-sessions
 
 # wayland-scanner is a tool which generates C headers and rigging for Wayland
 # protocols, which are specified in XML. wlroots requires you to rig these up
@@ -30,6 +33,14 @@ dwl.o: config.h xdg-shell-protocol.h
 
 dwl: xdg-shell-protocol.o
 
+install:
+	install -Dm 755 dwl $(BINDIR)/dwl
+	install -Dm 644 dwl.desktop $(DESKTOP)/dwl.desktop
+
+uninstall:
+	rm -f $(BINDIR)/dwl
+	rm -f $(DESKTOP)/dwl.desktop
+
 clean:
 	rm -f dwl *.o xdg-shell-protocol.h xdg-shell-protocol.c
 
diff --git a/dwl.desktop b/dwl.desktop
new file mode 100644
index 00000000..f2bb9c36
--- /dev/null
+++ b/dwl.desktop
@@ -0,0 +1,5 @@
+[Desktop Entry]
+Name=DWL
+Comment=dwm for Wayland
+Exec=/usr/local/bin/dwl
+Type=Application
